src/tsserver/nodeServer.ts

```diff
@@ -715,6 +715,38 @@ namespace ts.server {
                 this.constructed = true;
             }
 
+            public send(msg: protocol.Message) {
+                if (useNodeIpc) {
+                    if (msg.type === "event" && !this.canUseEvents) {
+                        if (this.logger.hasLevel(LogLevel.verbose)) {
+                            this.logger.info(`Session does not support events: ignored event: ${JSON.stringify(msg)}`);
+                        }
+                        return;
+                    }
+                    // TODO: Do we need any perf stuff here?
+                    // const msgText = formatMessage(msg, this.logger, this.byteLength, this.host.newLine);
+                    // perfLogger.logEvent(`Response message size: ${msgText.length}`);

```

Comment:
I think the answer here might be yes, or at least the portion that `formatMessage` does in `verboseLogging` mode as a side effect.